src/Login/index.test.js

```diff
@@ -1,15 +1,75 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { BrowserRouter as Router } from 'react-router-dom';
+import { MemoryRouter as Router, Switch, Route } from 'react-router-dom';
+import { MockedProvider } from '@apollo/client/testing';
+import AuthProvider from '../AuthProvider';
 import Login from './index';
+import QUERY_USERS from '../graphql/queries';
+import { HOME_PATH, LOGIN_PATH } from '../paths';
 
-it('renders the login page', () => {
-  render(<Router><Login /></Router>);
+const mockedUsersResponse = {
+  request: { query: QUERY_USERS },
+  result: {
+    data: {
+      users: [
+        {
+          username: 'User',
+          password: 'password',
+          id: 1,
+        },
+      ],
+    },
+  },
+};
+
+const mocks = [
+  mockedUsersResponse,
+  mockedUsersResponse,
+  mockedUsersResponse,
+];
+
+it('renders the login page', async () => {
```

Comment:
If it's not too much to ask, it would be great if you split the edge cases under this test into multiple ones that are small and have a specific goal, because this one covers a lot of cases which is good but makes it hard to read and will be even harder to maintain, for example you can split them like this and go even further with more tests if you want:
```
const App = () => (
  <MockedProvider mocks={mocks} addTypename={false}>
    <AuthProvider>
      <Router initialEntries={[LOGIN_PATH]}>
        <Switch>
          <Route exact path={HOME_PATH} component={() => <div>Home</div>} />
          <Route path={LOGIN_PATH} component={Login} />
        </Switch>
      </Router>
    </AuthProvider>
  </MockedProvider>
);

describe('Login page', () => {
  it('should render login form with initial state', async () => {
    render(<App />);
    // TODO
  });
  it('should not allow user to login when uername is invalid', async () => {
    render(<App />);
    // TODO
  });

  it('should not allow user to login when password is invalid', async () => {
    render(<App />);
    // TODO
  });

  it('should allow user to login when both uername and password are valid', async () => {
    render(<App />);
    // TODO
  });

  it('should {any other edge cases you can think of}', async () => {
    render(<App />);
    // TODO
  });
});
```